dapp/src/components/dialogs/GeneralInfoDialog.tsx

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) 2025 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+'use client';
+
+import {
+    Dialog,
+    DialogBody,
+    DialogContent,
+    DialogPosition,
+    Header,
+    KeyValueInfo,
+    TitleSize,
+    truncate,
+} from '@iota/apps-ui-kit';
+import { useCurrentAccount } from '@iota/dapp-kit';
+import Link from 'next/link';
+
+import { useNameRecord, useRegistrationNfts } from '@/hooks';
+import { formatDate } from '@/lib/utils/format/formatDate';
+import { normalizeNameInput } from '@/lib/utils/format/formatNames';
+
+import { Collapsible } from '../Collapsible';
+import { AvatarDisplay } from '../name-record/AvatarDisplay';
+
+interface GeneralInfoDialogProps {
+    name: string;
+    open: boolean;
+    setOpen: (bool: boolean) => void;
+}
+interface InfoLinks {
+    key: string;
+    value: string;
+    href: string;
+}
+
+export function GeneralInfoDialog({ name, open, setOpen }: GeneralInfoDialogProps) {
+    const account = useCurrentAccount();
+    const address = account?.address || '';
+
+    const { data: names } = useRegistrationNfts('name');
+    const { data: subnames } = useRegistrationNfts('subname');
+
+    const isSubname = !names?.find((n) => n.name === name);
+    const currentName = isSubname
+        ? subnames?.find((n) => n.name === name)
+        : names?.find((n) => n.name === name);
+
+    const currentNameId = currentName?.id ?? '';
+    const expirationTime = currentName?.expirationTimestampMs;
+
+    const { data: nameRecordData } = useNameRecord(name);
+    const targetAddress =
+        nameRecordData?.type === 'unavailable'
+            ? nameRecordData.nameRecord.targetAddress
+            : undefined;
```

Comment:
Simplified code + one less request to do
```suggestion
    const { data: nameRecordData } = useNameRecord(name);
    const { data: subnames } = useRegistrationNfts('subname');

    const nameRecord = nameRecordData as
        | Extract<NameRecordData, { type: 'unavailable' }>
        | undefined;
    const isNameSubname = isSubname(name);
    const { id, expirationTimestampMs } =
        (isNameSubname
            ? subnames?.find((n) => n.name === name)
            : { id: nameRecord?.nameRecord.nftId, ...nameRecord?.nameRecord }) ?? {};
```